vivarium_public_health/risks/distributions.py

```diff
@@ -384,9 +385,46 @@ def ppf(self, x: pd.Series, interpolation: bool = True) -> Union[np.ndarray, pd.
                        for name, dist in self._distributions.items()], axis=0)
 
 
-def get_distribution(risk: str, risk_type: str, builder) -> Union[BaseDistribution, EnsembleDistribution]:
+class CategoricalDistribution:
+    def __init__(self, exposure_data: pd.DataFrame, risk: str):
+        self.exposure_data = exposure_data
+        self._risk = risk
+
+    def setup(self, builder):
+        self.exposure = builder.value.register_value_producer(f'{self._risk}.exposure',
+                                                              source=builder.lookup.build_table(self.exposure_data))
+    @staticmethod
+    def _get_sorted_categories(data):
+        import re
+        new_keys = lambda key: [int(elem) if elem.isdigit() else elem.lower() for elem in re.split('([0-9]+)', key)]
+        return sorted(data, key=new_keys)
+
+    def ppf(self, x):
+        exposure = self.exposure(x.index)
+        # Get a list of sorted category names (e.g. ['cat1', 'cat2', ..., 'cat9', 'cat10', ...])
+        categories = self._get_sorted_categories([column for column in exposure if 'cat' in column])
+        sorted_exposures = exposure[categories]
+        try:
+            assert np.allclose(1, np.sum(sorted_exposures, axis=1))
+        except AssertionError:
+            raise MissingDataError('All exposure data returned as 0.')
+        exposure_sum = sorted_exposures.cumsum(axis='columns')
+        category_index = (exposure_sum.T < x).T.sum('columns')
+        return pd.Series(np.array(categories)[category_index], name=self._risk + '_exposure', index=x.index)
+
+
+def get_distribution(risk: str, risk_type: str, builder) -> Union[BaseDistribution, EnsembleDistribution, CategoricalDistribution]:
 
     distribution = builder.data.load(f"{risk_type}.{risk}.distribution")
+    if distribution == "dichotomous" or distribution == "polytomous":
```

Comment:
This is totally fine, but I tend to prefer something like `if distribution in ['dichotomous', 'polytomous']:` since it's easier to add additional cases to.